ql/src/java/org/apache/hadoop/hive/ql/ddl/table/drop/DropTableAnalyzer.java

```diff
@@ -51,7 +55,14 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     Table table = getTable(tableName, throwException);
     if (table != null) {
       inputs.add(new ReadEntity(table));
-      outputs.add(new WriteEntity(table, WriteEntity.WriteType.DDL_EXCLUSIVE));
+
+      boolean tableWithSuffix = (HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+          || HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED))
+        && AcidUtils.isTransactionalTable(table)
+        && table.getSd().getLocation().matches(V_TABLE_PATTERN);
```

Comment:
Might have issues with legacy tables with where the name (and location) ends with `_v1`.
Would it cause problems?
- Say I have an old table users_v2
- Switch on the `HIVE_ACID_CREATE_TABLE_USE_SUFFIX`
- Drop the table?

Reply:
- suffix starts with `dot` that is not allowed for table names in create statement;
- changed to table property `versioned` (added on create table)